src/main/java/Ui.java

```diff
@@ -0,0 +1,44 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class Ui {
+    protected static ArrayList<String> commandList = new ArrayList<>(Arrays.asList
+            ("todo", "deadline", "event", "mark", "unmark", "list", "bye", "delete"));
+    public void run(TaskList records) {
+        greet();
+        Scanner userInput = new Scanner(System.in);
+        Parser p = new Parser(records);
+        while (true) {
+            try {
+                String input = userInput.nextLine();
+                //Check for empty input (pressing enter)
+                if (input == "") {
+                    throw new EmptyInputException();
+                }
+                String[] inputArr = input.split(" ", 2);
+                //Check for invalid commands (first word)
+                boolean validCommand = commandList.contains(inputArr[0]);
+                if (!validCommand) {
+                    throw new InvalidCommandException();
+                }
+                p.handleInput(input, records);
+            }catch (EmptyInputException | InvalidCommandException exception) {
+                System.out.println("Input Error, please input a command within the list:" +
+                        "todo, deadline, event, mark, unmark, list, delete, bye.");
+            }
+        }
+    }
+
+    public static String format(String input) {
+        return "____________________________________________________________\n" +
```

Comment:
Can define the string as the private variable to make code cleaner

Reply:
Good suggestion, thanks